backend/common/src/main/java/ai/verta/modeldb/common/configuration/AppContext.java

```diff
@@ -12,9 +12,13 @@
 @Configuration
 public class AppContext implements ApplicationContextAware {
 
-  private static final AppContext appContext = new AppContext();
+  private static AppContext appContext;
   private ApplicationContext applicationContext;
 
+  public AppContext() {
+    appContext = this;
```

Comment:
Why are you overriding a static method in this constructor? This doesn't seem like a good pattern to me. If two callers construct one of these, you've now remove a context altogether.

Reply:
I want to access the app context class at the init test setup. If we create an App context like code before changes then I can't access application context in the Test Setup class. So I have introduced this context pattern so that it will take the same reference of class and I can access it there.

Reply:
I understand, but you need to find a better way. This also leaks the class reference out of the constructor which is a dangerous antipattern, since it could be accessed by another thread before it has finished being initialized.

Reply:
Ok I will check and will find a new good pattern..

Reply:
Reverted changes and added context from App.java class at the registry backend.